crates/primitives/src/transaction/mod.rs

```diff
@@ -1726,12 +1726,26 @@ impl TryFrom<reth_rpc_types::Transaction> for TransactionSignedEcRecovered {
     fn try_from(tx: reth_rpc_types::Transaction) -> Result<Self, Self::Error> {
         let signature = tx.signature.ok_or(ConversionError::MissingSignature)?;
 
+        let transaction: Transaction = tx.try_into()?;
+
         TransactionSigned::from_transaction_and_signature(
-            tx.try_into()?,
+            transaction.clone(),
             Signature {
                 r: signature.r,
                 s: signature.s,
-                odd_y_parity: signature.y_parity.ok_or(ConversionError::MissingYParity)?.0,
+                odd_y_parity: match transaction.tx_type() {
+                    TxType::Legacy => {
+                        signature.v -
+                            U256::from(if let Some(chain_id) = transaction.chain_id() {
+                                chain_id.saturating_mul(2).saturating_add(35)
+                            } else {
+                                27
+                            })
```

Comment:
this needs docs

this is basically the inverse of 

https://github.com/paradigmxyz/reth/blob/f6e1b0640cea24ed17efe2d4c09ceb591d6fa7cf/crates/primitives/src/transaction/signature.rs#L80-L80

wonder if t'd be useful to have this as a standalone function:

https://github.com/paradigmxyz/reth/blob/421888d22f038b0d304ac83efd600070f6c84ba4/crates/primitives/src/transaction/signature.rs#L102-L126

Reply:
Just added documentation here to give more context. Since these are two different types for signatures (rpc and primitives), I'm not sure one can make something super clean and better. I would say that for the moment we can leave it like that as it is only about the conversion and we can possibly see with a follow up PR if necessary, wdyt?

Reply:
A separate method makes sense as a followup

Reply:
Opened: https://github.com/paradigmxyz/reth/issues/7893